src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/StepAdaptationController.hpp

```diff
@@ -0,0 +1,304 @@
+        iDynTree::Transform m_footTransform; /**< transform of the next foot position. */
+
+        std::pair<bool, bool> m_feetStatus; /**< Current status of the feet. Left and Right. True is used
+                                            if the foot is in contact. */
+
+        bool m_isSolutionEvaluated{false}; /**< True if the solution is evaluated. */
+
+    public:
+        /**
+         * Constructor of step adaptation controller.
+         * @param inputSize size of the controlled input vector;
+         * @param numberOfAllConstraints number of equality and inequality constraints!
+         */
+        StepAdaptationController(const int& inputSize, const int& numberOfAllConstraints);
+
+        /**
+         * Compute the hessian matrix.
+         * Please do not call this function to update the hessian matrix! It can be set only once.
+         * @param zmpWeight weight of next step position term in cost function.
+         * @param dcmOffsetWeight weight of dcm offset term in cost function.
+         * @param sigmaWeight weight of next step timing term in cost function.
+         * @return true/false in case of success/failure.
+         */
+        bool computeHessianMatrix(const iDynTree::Vector2& zmpWeight, const iDynTree::Vector2& dcmOffsetWeight,
+                              const double& sigmaWeight);
+
+        /**
+         * Compute or update the linear constraints matrix(A) related to equality and inequality constraints(C<Ax<B)
+         * If the solver is already set the linear constraints matrix is updated otherwise it is set for
+         * the first time.
+         * @param currentDcmPosition This vector includes the current value of real DCM ;
+         * @param currentZmpPosition This vector includes the current value of stance foot position ;
+         * @param convexHullMatrix The convex hull matrix related to allowable next step position;
+         * @return true/false in case of success/failure.
+         */
+        bool computeConstraintsMatrix(const iDynTree::Vector2& currentDcmPosition, const iDynTree::Vector2& currentZmpPosition,
+                                  const iDynTree::MatrixDynSize& convexHullMatrix);
+
+        /**
+         * Compute or update the gradient
+         * @param zmpWeight weight of next step position term in cost function.
+         * @param dcmOffsetWeight weight of dcm offset term in cost function.
+         * @param sigmaWeight weight of next step timing term in cost function.
+         * @param zmpNominal vector of nominal values of next step position .
+         * @param dcmOffsetNominal vector of nominal values of dcm offset.
+         * @param sigmaNominal vector of nominal values of exp(w*steptiming) .
+         * @return true/false in case of success/failure.
+         */
+        bool computeGradientVector(const iDynTree::Vector2& zmpWeight, const iDynTree::Vector2& dcmOffsetWeight,
```

Comment:
I still see duplicated inputs here and there. Also, leave as ``public`` only those methods that are called from another class. Many of these methods are now used only by the ``StepAdaptationController`` itself.

Reply:
These functions all use only in solve function to have more clean solve function and so they need sometimes the same data! and I should specify them as private!

Reply:
Making them private is a first step. In addition, you could simplify the list of input parameters by using directly the class parameters, like ``m_zmpPositionWeight``.

Reply:
Thanks. Done!